sdk/daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/UpgradeCheckMain.scala

```diff
@@ -57,13 +57,21 @@ object UpgradeCheckMain {
       val validation = validator.validateUpgrade(archives.toList)
       Await.result(validation.value, Duration.Inf) match {
         case Left(err: Validation.Upgradeability.Error) =>
-          logger.error(s"Error while checking two DARs:\n${err.upgradeError.prettyInternal}")
-          sys.exit(1)
+          logger.error(s"Error while checking two DARs:\n${err.cause}")
+          1
         case Left(err) =>
           logger.error(s"Error while checking two DARs:\n${err.cause}")
-          sys.exit(1)
-        case Right(()) => ()
+          1
+        case Right(()) => 0
       }
     }
   }
+
+  def main(args: Array[String]) = sys.exit(check(args))
+}
+
+object UpgradeCheckMain {
+  def default: UpgradeCheckMain = {
```

Comment:
Nit: in the canton codebase they tend to call that "instance".

Also, could it be a val, or a lazy val? It seems to be quite stateless so I don't think you don't need a new instance each time.

Reply:
done